sql/functionregistry.go

```diff
@@ -21,6 +21,8 @@ type Function interface {
 }
 
 type (
+	// Function0 is a function with 0 arguments.
+	Function0 func() Expression
```

Comment:
Shall we use FunctionN() with no arguments (And ignore the unexpected arguments) for that function to avoid more code?

Reply:
either just `Function(...)` or `Function0()` + `FunctionN(...)`